### ISSUE TITLE
Data grid: `onCellClick` defined but not exposed

### ISSUE THREAD

**Issue by Janloe** [equinor/design-system#3839]
**Is your feature request related to a problem? Please describe.**

It is currently not possible to use the cell click handler `onCellClick()` in EDS Data grid.

E.g., the following code in the EdsDataGrid-component does not log to the console:

```javascript
onCellClick={() => console.log("cell clicked!")}
```
It is apparently passed to `TableBodyCell`, but is ignored from there.

**Describe the solution you'd like**

`onCellClick` would work similarly to `onRowClick`

**Describe alternatives you've considered**

For the time being, I have created a custom cell component with an `onClick` handler:

```javascript

cell: ({ row }: { row: Row<{ [key: string]: string }> }) => (
  <div onClick={() =>handleCellClick(Number(col.id))}>
    <Typography>{row.original[`${col.id}`]}</Typography>
  </div>
),
```

but this is suboptimal in my use case.

**Additional context**

Originally mentioned on slack here: https://equinor.slack.com/archives/CJT20H1B9/p1748436933075109

**Comment by pomfrida** [comment]
After discussion with design team, we're shifting approach to a standalone `ClickableCell` component that users can opt into, rather than built-in cell clicking. This puts accessibility responsibility on consumers while providing them with proper tools.
